### PR TITLE
Language Rearranging

### PR DESCRIPTION
**Pull request by tomlm**
- Add ILanguagePolicy as interface
- Make LanguagePolicy as default declarative class implementation
- Change LangaugeGenreation us use ILanguagePolicy
- Add MultiLanguageRecognizer with LanguagePolicy and map of language to IRecgonizer so you can have multi-language models active
- Add schema files
- Add Unit tests.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Rules/Recognizers/MultiLanguageRecognizer.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        public Task<RecognizerResult> RecognizeAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            if (!LanguagePolicy.TryGetValue(turnContext.Activity.Locale ?? string.Empty, out string[] policy))
+                policy = new string[] { string.Empty };
+
+            foreach (var option in policy)
+            {
+                if (this.Recognizers.TryGetValue(option, out IRecognizer recognizer))
+                {
+                    return recognizer.RecognizeAsync(turnContext, cancellationToken);
```
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Builder.Dialogs.Rules/Recognizers/MultiLanguageRecognizer.cs]: .ConfigureAwait(false);?

File: libraries/Microsoft.Bot.Builder.Dialogs.Rules/Recognizers/MultiLanguageRecognizer.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        public Task<T> RecognizeAsync<T>(ITurnContext turnContext, CancellationToken cancellationToken) where T : IRecognizerConvert, new()
+        {
+            if (!LanguagePolicy.TryGetValue(turnContext.Activity.Locale ?? string.Empty, out string[] policy))
+                policy = new string[] { string.Empty };
+
+            foreach (var option in policy)
+            {
+                if (this.Recognizers.TryGetValue(option, out IRecognizer recognizer))
+                {
+                    return recognizer.RecognizeAsync<T>(turnContext, cancellationToken);
```
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Builder.Dialogs.Rules/Recognizers/MultiLanguageRecognizer.cs]: .ConfigureAwait(false)?

File: libraries/Microsoft.Bot.Builder.Dialogs/LanguagePolicy.cs [hunk 3]
```diff
@@ -0,0 +1,4071 @@
...
+    /// <summary>
+    /// Language policy with fallback for each language as most specific to default en-us -> en -> default
+    /// </summary>
+    public class LanguagePolicy : Dictionary<string, string[]>, ILanguagePolicy
+    {
+        private static Lazy<IDictionary<string, string[]>> defaultPolicy = new Lazy<IDictionary<string, string[]>>(() =>
+        {
+            var cultureCodes = CultureInfo.GetCultures(CultureTypes.AllCultures).Select(c => c.IetfLanguageTag.ToLower()).ToList();
+            var policy = new Dictionary<string, string[]>();
+            foreach (var language in cultureCodes.Distinct())
```
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Builder.Dialogs/LanguagePolicy.cs]: Comment explaining what this chunk of code does pls

Comment (carlosscastro) [reply libraries/Microsoft.Bot.Builder.Dialogs/LanguagePolicy.cs]: Took me a bit to read and understand the purpose

File: libraries/Microsoft.Bot.Builder.Dialogs.Rules/Recognizers/MultiLanguageRecognizer.cs (hunk 2 above)
Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.Dialogs.Rules/Recognizers/MultiLanguageRecognizer.cs]: nope, not awaiting it I'm returning the task.

File: libraries/Microsoft.Bot.Builder.Dialogs.Rules/Recognizers/MultiLanguageRecognizer.cs (hunk 1 above)
Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.Dialogs.Rules/Recognizers/MultiLanguageRecognizer.cs]: same here, not awaiting it, returning the actual Task

**Pull request merged**